### PR TITLE
[DCE-161] Walking Skeleton - Test

### PR DESCRIPTION
**Pull request by ampats**
## Jira issue link
https://jira.pfizer.com/browse/DCE-161

## What type of PR is this? (check all applicable)

- [x] Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Optimization
- [x] Test
- [ ] Build
- [x] Documentation Update

## Description

Create testing project for dotnet application
- [x] Add a directory called test using xunit
- [x] Add the testing project to the solution
- [x] Update the README with instructions how to run the tests

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Color contrast tested?

## Added/updated tests?
- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

File: test/PracticeWebApi.Test/UnitTest1.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+namespace PracticeWebApi.Test;
+
+public class UnitTest1
+{
+    [Fact]
+    public void Test1()
+    {
+
```
Comment (SergiiVolodkoWorking) [inline test/PracticeWebApi.Test/UnitTest1.cs:5]: Could you please add a trivial assertion here? So the test is not empty and we have a way to see that the tests can fail if the assertion is false.

**Comment by Kterlecki** [comment]
Looks great! Can you add a link to PR in the top section of this pr

**Comment by ampats** [comment]
> Looks great! Can you add a link to PR in the top section of this pr

You mean a link to Jira issue (DCE-161 for this case)?

File: test/PracticeWebApi.Test/UnitTest1.cs (hunk 1 above)
Comment (ampats) [reply test/PracticeWebApi.Test/UnitTest1.cs:5]: I just added an assertion method with simple integer and a false condition.

File: PracticeWebApi.sln [hunk 2]
```diff
@@ -5,6 +5,10 @@ VisualStudioVersion = 17.0.31903.59
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebApi", "src\WebApi.csproj", "{8C3CBA82-994B-4F1F-B4A0-DC10B33E3D93}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "test", "test", "{51AC6806-323E-4B7F-96F9-7117B0B2E604}"
+EndProject
```
Comment (sethpthomas91) [inline PracticeWebApi.sln]: Look into:
```
dotnet sln remove
```
https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-sln

File: test/PracticeWebApi.Test/UnitTest1.cs [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
-    [Fact]
-    public void Test1()
-    {
+namespace PracticeWebApi.Test {
 
+    public class UnitTest1 {
+        [Fact]
+        public void Test1() {
+            int val = 1;
+            Debug.Assert(val != 1, " Value should not be 1.");
```
Comment (SergiiVolodkoWorking) [inline test/PracticeWebApi.Test/UnitTest1.cs]: Good stuff!
The important thing is that for testing we want to use testing liblaries, like the one standard for dot net. https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.assert.areequal
Can be replaced with something like
```
int val = 1;
Assert.AreEqual(val, 1);
```

Comment (ampats) [reply test/PracticeWebApi.Test/UnitTest1.cs]: When I used Assert.AreEqual() it threw some errors, probably because we used xUnit to create this testing project, while the first method is from the NUnit testing library, as I see here: https://learn.microsoft.com/en-us/visualstudio/test/getting-started-with-unit-testing?view=vs-2022&tabs=dotnet%2Cxunit (if I understand this correctly). So I used the Assert.Equal/.NotEqual method.

Comment (SergiiVolodkoWorking) [reply test/PracticeWebApi.Test/UnitTest1.cs]: You're absolutely correct! I missed that you're using xUnit. It should be just `Assert.Equal` indeed 👍

**Review by SergiiVolodkoWorking — APPROVED** [review]
Looks good!
